src/Microsoft.Identity.Web/Resource/JwtBearerMiddlewareDiagnostics.cs

```diff
@@ -2,43 +2,51 @@
 // Licensed under the MIT License.
 
 using System;
-using System.Diagnostics;
 using System.Threading.Tasks;
 using Microsoft.AspNetCore.Authentication.JwtBearer;
+using Microsoft.Extensions.Logging;
 
 namespace Microsoft.Identity.Web.Resource
 {
     /// <summary>
     /// Diagnostics for the JwtBearer middleware (used in Web APIs)
     /// </summary>
-    public class JwtBearerMiddlewareDiagnostics
+    public class JwtBearerMiddlewareDiagnostics : IJwtBearerMiddlewareDiagnostics
     {
+        private readonly ILogger _logger;
+
+        public JwtBearerMiddlewareDiagnostics(ILogger<JwtBearerMiddlewareDiagnostics> logger)
+        {
+            _logger = logger;
+        }
+
         /// <summary>
         /// Invoked if exceptions are thrown during request processing. The exceptions will be re-thrown after this event unless suppressed.
         /// </summary>
-        private static Func<AuthenticationFailedContext, Task> s_onAuthenticationFailed;
+        private Func<AuthenticationFailedContext, Task> s_onAuthenticationFailed;
 
         /// <summary>
         /// Invoked when a protocol message is first received.
         /// </summary>
-        private static Func<MessageReceivedContext, Task> s_onMessageReceived;
+        private Func<MessageReceivedContext, Task> s_onMessageReceived;
 
         /// <summary>
         /// Invoked after the security token has passed validation and a ClaimsIdentity has been generated.
         /// </summary>
-        private static Func<TokenValidatedContext, Task> s_onTokenValidated;
+        private Func<TokenValidatedContext, Task> s_onTokenValidated;
 
         /// <summary>
         /// Invoked before a challenge is sent back to the caller.
         /// </summary>
-        static Func<JwtBearerChallengeContext, Task> s_onChallenge;
+        private Func<JwtBearerChallengeContext, Task> s_onChallenge;
 
         /// <summary>
         /// Subscribes to all the JwtBearer events, to help debugging, while
         /// preserving the previous handlers (which are called)
         /// </summary>
         /// <param name="events">Events to subscribe to</param>
-        public static JwtBearerEvents Subscribe(JwtBearerEvents events)
+        public JwtBearerEvents Subscribe(
+            JwtBearerEvents events)
```

Comment:
you might want to remove the extra line?

Reply:
yes. thanks.